plugin-api/src/main/java/org/hyperledger/besu/plugin/data/BlockHeader.java

```diff
@@ -208,4 +208,12 @@ default Optional<Bytes32> getPrevRandao() {
    */
   @Unstable
   Optional<? extends Quantity> getExcessDataGas();
+
+  /**
+   * The data_gas_used of this header.
+   *
+   * @return The data_gas_used of this header.
+   */
+  @Unstable
+  long getDataGasUsed();
```

Comment:
It is an unsigned int64, and there is another PR that is addressing the long/DataGas mismatch:  

https://github.com/hyperledger/besu/pull/5654

Reply:
That PR has worse performance implications than staying with longs, there was about a 20% evm speed improvement when I moved to longs for gas away from the bespoke type the PR is proposing. 

Even though it's an unsigned 64 bit long there are `Long` methods that correctly handle comparisons and multiplication/division.  I would recommend settling on the primitive type.

Reply:
understood, but that gives me readability concerns. when a reader sees a long, they assume it is signed, and would naturally interact with it as signed. Is there a way we can use this as a primitive but encode the "unsigned" semantics somehow? extend Long maybe?

Reply:
Use https://checkerframework.org/manual/#signedness-checker and annotate it with `org.checkerframework.errorprone.checker.signedness.qual.Unsigned`.  We should look into running the signedness checker as part of the build, since unsigned longs are very much a part of the EVM.

Reply:
I have reverted it to use longs, and will introduce checker framework in a later branch.  

https://github.com/hyperledger/besu/issues/5696